app/models/geometry.rb

```diff
@@ -85,16 +90,19 @@ def in_slop?(f)
 end
 
 class Float
-  def in?(s, e, inclusive = true)
+  def in?(starting, ending, inclusive = true)
     if inclusive
-      self >= s && self <= e
+      self >= starting && self <= ending
     else
-      self > s && self < e
+      self > starting && self < ending
     end
   end
 end
 
+# Use a struct, for easy equality checks
 Pixel = Struct.new(:x, :y) do
+  include Target
```

Comment:
the other version of `Pixel` was overriding this version (and throwing a warning about replacing an existing class). But the struct version has equality comparisons, and the geometry module seems to expect that. So I removed the `Pixel` class, and just included the `Target` methods in the struct instead.